quipucords/scanner/network/connect.py

```diff
@@ -30,6 +31,77 @@
 logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
 
 
+# The ConnectTaskRunner creates a new ConnectResultCallback for each
+# credential it tries to connect with, and the ConnectResultCallbacks
+# all forward their information to a single ConnectResultStore.
+class ConnectResultStore(object):
+    """This object knows how to record and retrieve connection results."""
+
+    def __init__(self, scan_task, conn_results):
+        """Get the unique ConnectionResult object for this scan."""
+        self.scan_task = scan_task
+        self.conn_results = conn_results
+
+        source = scan_task.source
+
+        with transaction.atomic():
+            conn_result = conn_results.results.filter(
+                source__id=source.id).first()
+            if conn_result is None:
+                conn_result = ConnectionResult(
+                    scan_task=scan_task, source=source)
+                conn_result.save()
+                conn_results.results.add(conn_result)
+                conn_results.save()
+        self.conn_result = conn_result
+
+        hosts = set((host_range.host_range
+                     for host_range in source.hosts.all()))
+        for sys_result in conn_result.systems.all():
+            hosts.discard(sys_result.name)
+        self._remaining_hosts = hosts
+
+        if scan_task.systems_count is None:
+            scan_task.systems_count = len(hosts)
```

Comment:
I don't think this is the number of hosts. Just the number of entries provided. If its a range it will not capture the number in the range.
Below is the code from `rho` that pulled in the Ansible info to determine sys count: https://github.com/quipucords/rho/blob/master/rho/host_discovery.py#L127

Reply:
Fixed, thank you!